### PR TITLE
Make things support multi-owner

### PR DESCRIPTION
**Pull request by upupzealot**
When a model has two or more "belongsTo"  relations, $owner will check
them all, instead of only checking the first one.

### CODE REVIEW

**Comment by 0candy** [comment]
Thank you for submitting a PR. Please fix the long line: 
```/home/travis/build/strongloop/loopback/common/models/role.js
  221:1  error  Line 221 exceeds the maximum line length of 100  max-len
```

Also, please add a unit test to verify your changes.

**Comment by upupzealot** [comment]
@0candy this is strange, because I don't change this line, it's original [line 222](https://github.com/strongloop/loopback/blob/1559db2ca3f5cf75606a09a806428140e9c967d6/common/models/role.js#L222) in role.js

About the test, I will do that, but not sure how.
Should I add a test file under the `/test` folder or should I modify the `role.test.js` which already exists?

**Comment by 0candy** [comment]
@upupzealot My bad, don't worry about the long line, it was due to someone else's change. As for the unit test, you can just add a test case to `role.test.js`. Thanks!

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by upupzealot** [comment]
any progress?

**Comment by ahmed-abdulmoniem** [comment]
++++1

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by 0candy** [comment]
@bajtos Please review.

**Comment by bajtos** [comment]
I am not familiar with this part of our codebase, I believe @raymondfeng and/or @ritch are more suitable to review this patch.

BTW the code needs rebasing on top of the current master, `test/role.test.js` was changed quite a bit recently to fix intermittent test failure.

**Comment by VictorAlbertos** [comment]
Hi :)

Are you still planing to integrate this feature?

Thanks.

**Comment by ahmed-abdulmoniem** [comment]
+1

**Comment by upscreen** [comment]
Good day

When can we expect this feature to be available in the master?

Regards.

**Comment by 0candy** [comment]
@upupzealot Could you please rebase your code?

@raymondfeng Could you please review?

**Comment by upupzealot** [comment]
@0candy conflict in the test. How to solve this?

**Comment by upupzealot** [comment]
Did "Allow edits from maintainers", hope it would help somehow.

**Comment by 0candy** [comment]
Hi @upupzealot , I rebased onto your branch, however some of the unit tests are failing. Could you please take a look? Thanks!

File: common/models/role.js [hunk 1]
```diff
@@ -210,26 +210,42 @@ module.exports = function(Role) {
         if (callback) callback(null, matches(ownerId, userId));
         return;
       } else {
```
Comment (pierreclr) [inline common/models/role.js]: That way you allow multiple owner only when it's not base on the `userId` / `owner` field ... see https://github.com/strongloop/loopback/pull/3106/files for my implementation

Comment (upupzealot) [inline common/models/role.js]: @pierreclr Hi, for some reason I'm no longer maintaining this pr, it's glad to see somebody else fulfilled the request for mult-owner. Hope to see this be merged into the matser soon.

**Comment by bajtos** [comment]
Closing in favour of #3106

**Pull request closed without merging**